source/mir/math/stat.d

```diff
@@ -420,6 +424,157 @@ unittest
     assert(float.max.repeat(3).hmean!(double, "fast").approxEqual(float.max));
 }
 
+private template centerImpl(F)
+{
+    import mir.ndslice.slice: Slice, SliceKind;
+
+    @safe pure nothrow @nogc
+    auto centerImpl(Iterator, size_t N, SliceKind kind)(
+        Slice!(Iterator, N, kind) slice, F average)
+    {
+        return slice - average;
+    }
+}
+
+/++
+Centers `r`, which must be a finite iterable.
+
+By default, `r` is centered by the mean. A custom function may also be provided
+using `centralTendency`.
+
+Returns:
+    The elements in the range r with the average subtracted from them.
++/
+
+/// ditto
+template center(alias centralTendency = mean!(Summation.appropriate))
+{
+    /++
+    Params:
+        r = range
+    +/
+    @safe pure nothrow
+    auto center(Range)(Range r)
+        if (isIterable!Range)
+    {
+        return centerImpl!(sumType!Range)(r, centralTendency(r));
```

Comment:
`slice - centralTendency(slice) `

Reply:
better to use `move`
```
import core.lifetime: move;
improt mir.ndslice.topology: vmap;
import mir.ndslice.internal: RightOp;
auto m = centralTendency(r);
alias T = typeof(m);
return r.move.vmap(RightOp!(op, ImplicitlyUnqual!T)(m));
```

Reply:
EDIT: RightOp -> LeftOp

Reply:
`op` would be passed to `template center(...)`, right?

Reply:
Yeah, `template center(alias centralTendency = mean!(Summation.appropriate), string op = "-")` works.

Reply:
ah, we can hardcoded it to `LeftOp!("-", ImplicitlyUnqual!T) `